# Controller 1
rocr_controller:
  type        : "diff_drive_controller/DiffDriveController"
  left_wheel  : 'joint_left_rim'
  right_wheel : 'joint_right_rim'
  publish_rate: 50.0               # default: 50
  pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]

  # Wheel separation and diameter. These are both optional.
  # diff_drive_controller will attempt to read either one or both from the
  # URDF if not specified as a parameter
  wheel_separation : 1.0
  wheel_radius : 0.3

  # Wheel separation and radius multipliers
  wheel_separation_multiplier: 1.0 # default: 1.0
  wheel_radius_multiplier    : 1.0 # default: 1.0

  # Velocity commands timeout [s], default 0.5
  cmd_vel_timeout: 0.25

  # Base frame_id
  base_frame_id: base_footprint #default: base_link

  # Velocity and acceleration limits
  # Whenever a min_* is unspecified, default to -max_*
  linear:
    x:
      has_velocity_limits    : true
      max_velocity           : 1.0  # m/s
      min_velocity           : -0.5 # m/s
      has_acceleration_limits: true
      max_acceleration       : 0.1  # m/s^2
      min_acceleration       : -0.05 # m/s^2
      has_jerk_limits        : true
      max_jerk               : 5.0  # m/s^3
  angular:
    z:
      has_velocity_limits    : true
      max_velocity           : 1.7  # rad/s
      has_acceleration_limits: true
      max_acceleration       : 0.1  # rad/s^2
      has_jerk_limits        : true
      max_jerk               : 2.5  # rad/s^3

# Controller 2
controller_left_arm: 
    type: effort_controllers/JointPositionController
    joint: joint_left_arm
    pid: {p: 500.0, i: 20.5, d: 4.0}

# Controller 3
controller_right_arm: 
    type: effort_controllers/JointPositionController
    joint: joint_right_arm
    pid: {p: 500.0, i: 20.5, d: 4.0}